### PR TITLE
Make SEV-SNP the default attestation variant on GCP

### PR DESCRIPTION
**Pull request by thomasten**
### Context
SNP on GCP is GA

### Proposed change(s)
- make attestation variant gcp-sev-snp the default on GCP

### Checklist
- [ ] Run the E2E tests that are relevant to this PR's changes
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

**Comment by thomasten** [comment]
@msanft does this TODO need to be resolved before we can make SNP the default?
https://github.com/edgelesssys/constellation/blob/3f6e7f9c532d8df0ae2875f6bb95cca7063dada7/.github/workflows/on-release.yml#L158-L166

**Comment by msanft** [comment]
> @msanft does this TODO need to be resolved before we can make SNP the default?
> 
> https://github.com/edgelesssys/constellation/blob/3f6e7f9c532d8df0ae2875f6bb95cca7063dada7/.github/workflows/on-release.yml#L158-L166

Need to talk to the business side to clarify this, but I think it should not be too much work altogether.

I'll have a look at this PR generally on Monday. My mind isn't fresh enough anymore today :smile:

**Comment by elchead** [comment]
should we change it here too?

https://github.com/edgelesssys/constellation/blob/74e0f44230f1d1f79c2a2f0c5a98a09f33eb990d/.github/workflows/e2e-test-release.yml#L407

https://github.com/edgelesssys/constellation/blob/74e0f44230f1d1f79c2a2f0c5a98a09f33eb990d/.github/workflows/e2e-test-weekly.yml#L416

https://github.com/edgelesssys/constellation/blob/74e0f44230f1d1f79c2a2f0c5a98a09f33eb990d/.github/workflows/e2e-test-weekly.yml#L494

is this done?
https://github.com/edgelesssys/constellation/blob/9cd11842442dab3864c07b8788c4836e0ee25401/.github/workflows/on-release.yml#L162

let's switch the order here:
https://github.com/edgelesssys/constellation/blob/913b09aeb85967f673a4b68882f54f7d3043754b/terraform-provider-constellation/internal/provider/shared_attributes.go#L34

**Review by daniel-weisse — APPROVED** [review]

File: docs/docs/overview/clouds.md [hunk 1]
```diff
@@ -34,15 +34,14 @@ Thus, the Azure closed-source firmware becomes part of Constellation's trusted c
 
 ## Google Cloud Platform (GCP)
 
-The [CVMs Generally Available in GCP](https://cloud.google.com/confidential-computing/confidential-vm/docs/confidential-vm-overview#amd_sev) are based on AMD SEV but don't have SNP features enabled.
-CVMs with [SEV-SNP enabled are in public preview](https://cloud.google.com/confidential-computing/confidential-vm/docs/confidential-vm-overview#amd_sev-snp). Regarding (3), with their SEV-SNP offering Google provides direct access to remote-attestation statements.
+The [CVMs Generally Available in GCP](https://cloud.google.com/confidential-computing/confidential-vm/docs/confidential-vm-overview#technologies) are based on AMD SEV-ES or SEV-SNP.
+Regarding (3), with their SEV-SNP offering Google provides direct access to remote-attestation statements.
```
Comment (msanft) [inline docs/docs/overview/clouds.md:38]:
```suggestion
Regarding (3), with their SEV-SNP offering Google provides direct access to attestation statements.
```
nit

**Review by msanft — APPROVED** [review]
LGTM. Marketplace images should not be critical for this PR, we'll just have to add them until we release a version containing the SEV-SNP default.

**Review by elchead — APPROVED** [review]

**Pull request merged**